src/twisted/internet/kqreactor.py

```diff
@@ -16,12 +16,20 @@
    kqreactor.install()
 """
 
+import sys
+
+# For mypy's benefit
+assert not sys.platform.startswith("linux")
```

Comment:
I can't say I really like this, since it assumes that no Linux version supports kqueue, and I don't know if that's right…

I tried this for the epollreactor.py on macOS, but it doesn't suppress the error:

```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    import sys
    assert sys.platform != "darwin"
```

So I can't figure out how to only make the assertion for mypy.

Reply:
I think that's a pretty safe bet.  kqueue() started on FreeBSD ( https://people.freebsd.org/~jlemon/papers/kqueue.pdf ), and was later ported to other BSD operating systems including Darwin.

However, I don't like to start putting the equivalent of the old C way of doing platform specific #ifdef's if we don't have to, to avoid breaking some oddball platform.

Is it possible to appease mypy by doing something like:

```
import select
assert 'epoll' in dir(select)
```

or:

```
import select
assert 'kqueue' in dir(select)
```

On Linux, `dir(select)` has `epoll`, but not `kqueue`,
while on MacOS, `dir(select)` has `kqueue` but not `epoll`

Reply:
@rodrigc: nah, didn't work :-(

I guess we can leave this `sys.platform` hack in place for now.  If we were hypothetically to end up with `kqueue` on Linux, this assertion would trip in CI and maybe by then we'll have better options.

Reply:
Turns out even that wasn't working, and I was about to give up, but @graingert to the rescue with a suggestion to use `getattr` (though he was talking about #1427, but I'll take it!).